controllers/k8ssandra/k8ssandracluster_controller_test.go

```diff
@@ -2216,3 +2218,177 @@ func injectContainersAndVolumes(t *testing.T, ctx context.Context, f *framework.
 	require.NoError(err, "failed to delete K8ssandraCluster")
 	f.AssertObjectDoesNotExist(ctx, t, dc1Key, &cassdcapi.CassandraDatacenter{}, timeout, interval)
 }
+
+// createMultiDcDseCluster verifies that the CassandraDatacenters are created in the expected
+// order, based on the workload of each DC.
+func createMultiDcDseCluster(t *testing.T, ctx context.Context, f *framework.Framework, namespace string) {
+	require := require.New(t)
+
+	clusterName := "cluster-multi"
+	kc := &api.K8ssandraCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      clusterName,
+		},
+		Spec: api.K8ssandraClusterSpec{
+			Cassandra: &api.CassandraClusterTemplate{
+				ServerType: "dse",
+				Datacenters: []api.CassandraDatacenterTemplate{
+					{
+						Meta: api.EmbeddedObjectMeta{
+							Name: "dc1",
+						},
+						K8sContext: f.DataPlaneContexts[0],
+						Size:       3,
+						DatacenterOptions: api.DatacenterOptions{
+							ServerVersion: "6.8.17",
+							StorageConfig: &cassdcapi.StorageConfig{
+								CassandraDataVolumeClaimSpec: &corev1.PersistentVolumeClaimSpec{
+									StorageClassName: &defaultStorageClass,
+								},
+							},
+							DseWorkloads: &cassdcapi.DseWorkloads{
+								SearchEnabled: true,
+							},
+						},
+					},
+					{
+						Meta: api.EmbeddedObjectMeta{
```

Comment:
Shouldn't we add more dcs and also test a dc with Cassandra and a dc with Graph workloads?